teuthology/misc.py

```diff
@@ -550,27 +550,31 @@ def copy_file(from_remote, from_path, to_remote, to_path=None):
     ])
 
 
-def move_file(remote, from_path, to_path, sudo=False):
+def move_file(remote, from_path, to_path, sudo=False, preserve_perms=True):
     """
     Move a file from one path to another on a remote site
 
-    The file needs to be stat'ed first, to make sure we
-    maintain the same permissions
-    """
-    args = []
-    if sudo:
-        args.append('sudo')
-    args.extend([
-        'stat',
-        '-c',
-        '\"%a\"',
-        to_path
-    ])
-    proc = remote.run(
-        args=args,
-        stdout=StringIO(),
-    )
-    perms = proc.stdout.getvalue().rstrip().strip('\"')
+    If preserve_perms is true, the contents of the destination file (to_path,
+    which must already exist in this case) are replaced with the contents of the
+    source file (from_path) and the permissions of to_path are preserved. If
+    preserve_perms is false, to_path does not need to exist, and is simply
+    clobbered if it does.
+    """
+    if preserve_perms:
+        args = []
+        if sudo:
+            args.append('sudo')
+        args.extend([
+            'stat',
+            '-c',
+            '\"%a\"',
```

Comment:
Are the backslashes for the further shell escape?  This is what "run.Raw()" is for, but I also don't think the quotes are necessary:

>>> r.run(args=['echo', '%a'])
2017-03-20 09:13:46,118.118 INFO:teuthology.orchestra.run.mira001:Running: 'echo %a'
2017-03-20 09:13:46,302.302 INFO:teuthology.orchestra.run.mira001.stdout:%a

Edit: I see you were just moving code.  Still.